File: javascript/day04/part1.js
const data = require('../file-reader.js')
    .readFile('input.txt', '\n', String)
    .map(s => s.split(',').map(s => s.split('-').map(Number)).map(a => ({ s: a[0], e: a[1]})))

function isContained(pair) {
    const [a, b] = pair
    return a.s <= b.s && a.e >= b.e || b.s <= a.s && b.e >= a.e
}

const sum = data.reduce((sum, pair) => sum + isContained(pair), 0)
console.log(sum)

File: python/day04/part1.py
t = 0

for line in open('input.txt'):
    a, b, x, y = map(int, line.replace(",", "-").split("-"))
    if a <= x and b >= y or x <= a and y >= b:
        t += 1

print(t)
assert t == 444, 'Incorrect result'
